cve_bin_tool/parsers/javascript.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import json
+
+from cve_bin_tool.parsers import Parser
+from cve_bin_tool.util import ProductInfo, ScanInfo
+
+
+class JavascriptParser(Parser):
+    def __init__(self, cve_db, logger):
+        self.cve_db = cve_db
+        self.logger = logger
+        self.filename = ""
+
+    def find_vendor(self, product, version):
+        """Find vendor for Javascript product"""
+        if version == "*":
+            return None
+        vendor_package_pair = self.cve_db.get_vendor_product_pairs(product)
+        vendorlist: list[ScanInfo] = []
+        if vendor_package_pair != []:
+            # To handle multiple vendors, return all combinations of product/vendor mappings
+            for v in vendor_package_pair:
+                vendor = v["vendor"]
+                file_path = self.filename
+                # Tidy up version string
+                if "^" in version:
+                    version = version[1:]
+                self.logger.debug(f"{file_path} {product} {version} by {vendor}")
+                vendorlist.append(
+                    ScanInfo(ProductInfo(vendor, product, version), file_path)
+                )
+            return vendorlist if len(vendorlist) > 0 else None
+        return None
+
+    def run_checker(self, filename):
+        """Process package-lock.json file and extract product and dependency details"""
+        self.filename = filename
+        fh = open(self.filename)
```

Comment:
I know you're just moving stuff, but while we're at it can we change this to use the `with` construction for files so it handles the auto-close and checking to make sure it opened before we call the json parser?  

e.g. this line would be
```python
with fh as open(self.filename):
```
And you'd have to indent the rest to go with that.  

More info: https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files

Reply:
(You might also want to do it with the json load, but I don't know offhand how that one handles failures so it might not be as important.)